analytics.ts

```diff
@@ -0,0 +1,159 @@
+	const writeConsentValueToSettings = (option: Consent) => {
+		const input = prompt(option === OPT_IN ? optInConfirmationPrompt : optOutConfirmationPrompt);
+		if (input && /^y(es)?$/i.test(input)) {
+			return pipe(
+				readJsonFile<Settings.t>(settingsFilePath),
+				map((settingsContent: Settings.t) => {
+					settingsContent.consent = option;
+					return settingsContent;
+				}),
+				chain(writeJsonFile(settingsFilePath)),
+				map(() =>
+					option === OPT_IN
+						? 'You have successfully opted-in to sharing anonymous usage analytics'
+						: 'You have successfully opted-out from sharing anonymous usage analytics'
+				),
+			);
+		} else {
+			return taqResolve('');
+		}
+	};
+
+	const isCIRun = () => env.get('CI') !== undefined;
+
+	const isTestRun = () => env.get('TEST') !== undefined;
+
+	const promptForConsent = (): Future<TaqError.t, string> => {
+		const input = prompt(consentPrompt);
+		const option = input && /^y(es)?$/i.test(input) ? optInAnalyticsFirstTime : optOutAnalyticsFirstTime;
+		return pipe(
+			mkdir(settingsFolder),
+			chain(() => option()),
+		);
+	};
+
+	const allowTracking = (): Future<TaqError.t, boolean> => {
+		if (
+			isCIRun()
+			|| isTestRun()
+			|| inputArgs.includes('--version')
+			|| inputArgs.includes('--build')
+			|| inputArgs.includes('testFromVsCode')
+		) {
+			return taqResolve(false);
+		}
+		return pipe(
+			// If path/file exists,
+			// then the key 'consent' will be present because this is the 1st iteration of the settings file,
+			// and the taq settings directory in home will exist as well.
+			doesPathExist(settingsFilePath),
+			chainRej(() => promptForConsent()),
+			chain(() => readJsonFile<Settings.t>(settingsFilePath)),
+			map((settings: Settings.t) => settings.consent === OPT_IN),
+		);
+	};
+
+	// No-operation
+	// noop: () -> void
+	const noop = () => {};
+
+	const sendEvent = (
+		taq_version: string,
+		taq_ui: 'CLI' | 'VSCode',
+	): Future<TaqError.t, void> => {
+		if (taq_ui === 'VSCode') return resolve(noop()); // Disable for VSCode for now
+		return pipe(
+			allowTracking(),
+			chain((allowTracking: boolean) => {
+				if (!allowTracking) {
+					return resolve(noop());
+				}
+
+				const measurement_id = 'G-8LSQ6J7P0Q';
+				const api_secret = '3aHoMp2USE21ZPmAVTI1Lg';
+
+				const currentTime = new Date();
+				const taq_timestamp = currentTime.toDateString() + ', ' + currentTime.toTimeString();
+
+				return pipe(
+					attemptP<TaqError.t, string>(() => getMachineId()),
+					chain(client_id =>
+						attemptP<TaqError.t, void>(() =>
+							fetch(
+								`https://www.google-analytics.com/mp/collect?measurement_id=${measurement_id}&api_secret=${api_secret}`,
+								{
+									method: 'POST',
+									body: JSON.stringify({
+										client_id,
+										events: [{
+											name: 'taq_task_executed',
+											params: {
+												taq_version,
+												taq_ui,
+												taq_timestamp,
+												taq_os: Deno.build.os,
```

Comment:
Could we use dependency injection to get Deno.build.os please? We'll need this when writing tests.